src/main/java/app/sami/languageWeb/contractRequest/ContractRequestController.java

```diff
@@ -28,4 +35,25 @@ public ContractRequestDto renderDecision(@PathVariable Long contractRequestId,
         return ContractRequestMapper.toContractRequestDto(contractRequestService
                 .renderContractRequestDecision(status.getStatus(), contractRequestId, subject));
     }
+
+    @GetMapping("/contract-requests")
+    public ContractRequestsDto getContractRequests(@RequestParam Long requestId,
+                                                   @RequestJwtSubject UUID subject){
+        Request request = requestRepository.findById(requestId).orElseThrow(NotFoundException::new);
```

Comment:
You are doing 2 calls to the DB whereas you could filter them in memory as the number of contract request will always be very low